### ISSUE TITLE
ReduceLROnPlateau scheduler, checking multiple times per epoch

### ISSUE THREAD

**Issue by baalliso** [PyTorchLightning/pytorch-lightning#2316]
I'm training a model on a very large amount of data, where we evaluate multiple times per epoch. The best performing and most general learning rate scheduler for our domain is the ReduceLROnPlateau scheduler.

I'd like to use Lightning to allow relatively simple multi-gpu training. However, it looks like when you use the ReduceLROnPlateau scheduler, you can either only evaluate at the end of an epoch (not frequently enough) or every training batch where you only have access to training metrics not validation scores.

Is there a simple way to accomplish what I want? Thanks!

**Comment by SkafteNicki** [comment]
It was discussed here:
https://github.com/PyTorchLightning/pytorch-lightning/issues/1156

Conclusion:
Set `val_check_interval` to how often you want to evaluate (lets say every 50 step)
`trainer = Trainer(val_check_interval=50)`
Initialize scheduler with the frequency argument. It should be a **larger** number than your validation frequency:
```
scheduler = {
    'schduler': ReduceLROnPlateau(optimizer, mode, factor, patience),
    'interval': 'step',
    'frequency': 100
}
```
This will run validation loop every 50 step and call the scheduler every 100 step.

**Comment by cowwoc** [comment]
https://github.com/PyTorchLightning/pytorch-lightning/pull/7032/files#r632921255 contains a useful hint:

> If someone wants it to align it with validation, they can set scheduler['frequency'] = check_val_every_n_epoch.
